### DISCUSSION TITLE
Installation

### DISCUSSION THREAD

**Discussion by inmarx** [thkl/hap-homematic#610] [General]
Hallo,
Ich bin Neuling, ich versuche verzweifelt das Add-on zu installieren. Über die GUI bekomme ich es nicht hin, per ssh bekomme ich folgende Meldung:

root@ccu3-webui:/usr/local/addons/hap-homematic# npm i hap-homematic
npm WARN tar zlib: incorrect data check
npm WARN tarball tarball data for tr46@https://registry.npmjs.org/tr46/-/tr46-0.0.3.tgz (sha512-N3WMsuqV66lT30CrXNbEjx4GEwlow3v6rr4mCcv6prnfwhS01rkgyFdjPNBYd9br7LpXV1+Emh01fHnq2Gdgrw==) seems to be corrupted. Trying again.
npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
npm WARN tarball tarball data for tr46@https://registry.npmjs.org/tr46/-/tr46-0.0.3.tgz (sha512-N3WMsuqV66lT30CrXNbEjx4GEwlow3v6rr4mCcv6prnfwhS01rkgyFdjPNBYd9br7LpXV1+Emh01fHnq2Gdgrw==) seems to be corrupted. Trying again.
npm WARN tar zlib: incorrect data check
npm WARN tarball tarball data for googleapis@https://registry.npmjs.org/googleapis/-/googleapis-110.0.0.tgz (sha512-k6de3PGsdFEBULMiFwPYCKOBljDTDvHD3YGe/OFqe8Ot0lYQPL8QV1qjxjrPWiE/Ftf0Ar2v4DNES66jLfSO7w==) seems to be corrupted. Trying again.
npm WARN tarball tarball data for googleapis@https://registry.npmjs.org/googleapis/-/googleapis-110.0.0.tgz (sha512-k6de3PGsdFEBULMiFwPYCKOBljDTDvHD3YGe/OFqe8Ot0lYQPL8QV1qjxjrPWiE/Ftf0Ar2v4DNES66jLfSO7w==) seems to be corrupted. Trying again.
npm notice 
npm notice New major version of npm available! 8.19.2 -> 9.4.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v9.4.0
npm notice Run npm install -g npm@9.4.0 to update!
npm notice 
npm ERR! code ENOENT
npm ERR! syscall stat
npm ERR! path /tmp/npm-cache/_cacache/content-v2/sha512/37/75/8cb2ea95eba953df40ab5cd6c48f1e06130968c37bfaaebe2609cbfaa6b9dfc214b4d6b920c857633cd05877d6ebecba57575f849a1d357c79ead86760af
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, stat '/tmp/npm-cache/_cacache/content-v2/sha512/37/75/8cb2ea95eba953df40ab5cd6c48f1e06130968c37bfaaebe2609cbfaa6b9dfc214b4d6b920c857633cd05877d6ebecba57575f849a1d357c79ead86760af'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/npm-cache/_logs/2023-01-26T15_03_21_639Z-debug-0.log
root@ccu3-webui:/usr/local/addons/hap-homematic# 

Habt ihr eine Idee?
Danke!

**Comment by gerd2005** [comment]
Was passiert denn in der GUI? Es dauert tatsächlich einige Minuten bis der Button „Homekit“ in der Systemsteuerung erscheint.

**Reply by KlFeigling1** [reply]
Hallo Zusammen, ich habe heute das Plugin auf meine CCU3 installiert und es wurde bei mir auch erst nach einem Neustart der CCU3 angezeigt. Somit wirklich 5min warten bis es angezeigt wird. 

Nach dem es endlich angezeigt wurde, kann ich leider keine Brige zu meinem Homkit hinzufügen. Gibt es da noch ein Tipp von euch? 
Am Iphone kommt immer die Meldung, dass keine Bridge gefunden wird, wenn ich den Barcode einscanne. Habe es schon mit Geräten und auch ohne Geräten versucht. 

- Alle Ports der CCU3 freigegeben 
- HomePod Mini mit  Version 16.3.2
- Iphone Version mit Version 16.3
- Raspberrymatic mit Version: 3.67.10.20230225

Hoffe ihr könnt mir Helfen oder wenigsten sagen, ob es überhaut mit den aktuellen IOS-Versionen funktioniert.

**Reply by dkrizic** [reply]
Sind das iPhone und die CCU3 im selben Netz oder bildet das WLAN ein eigenes NAT?

**Reply by KlFeigling1** [reply]
Beide sind im selben Netzwerk

**Reply by blackbasket** [reply]
Also erstmal die Frage, warum dein iPhone auf der unsicheren iOS-Version 16.3 ist? ;)

Aber zum Thema: Ist HomeKit ggf. auf der neuen Version? Da konnte man die QR-Codes IMHO nur noch für Matter scannen. Was passiert bei manueller Eingabe?

**Reply by KlFeigling1** [reply]
Hab nochmal nachgeschaut und habe Version 16.3.1 auf meinem Handy installiert. Leider klappt es auch nicht, wenn ich den Code manuell eingebe. Habe das Plugin schon deinstalliert und nochmal installiert. Aber es half nicht.

**Reply by blackbasket** [reply]
Ok... ich bin der Meinung, dass ich die Bridges sowieso vor dem Code ausgewählt habe... du tippst auf "weitere Optionen"?

Kannst du bitte einmal einen Screenshot deiner FW-config und der HAP-Bridges posten?

**Reply by jb-home** [reply]
Sind die Instanzen denn Veröffentlicht?

**Reply by KlFeigling1** [reply]
Also die Instanzen sind veröffentlicht und bei den Screenshots bin ich mir unsicher was man hier veröffentlichen darf. Nicht das ihr alle auf meine Geräte zugreifen könnt. Mache nachher mal ein paar Screenshots, blende aber die Codes aus.

**Reply by KlFeigling1** [reply]
Ich habe nun die alternative mit Node-Red ausprobiert und diese funktioniert auf anhieb und Kinderleicht. Somit bleibe ich erstmal bei der Lösung. Trotzdem danke für eure Hilfe.
